app/views/rubygems/show.html.erb

```diff
@@ -4,7 +4,11 @@
 <% content_for :head do %>
   <%= auto_discovery_link_tag(:atom, rubygem_versions_path(@rubygem, format: "atom"), {title: "#{@rubygem.name} Version Feed"}) %>
   <% if @rubygem.versions.any? && @latest_version.indexed %>
-    <link rel="canonical" href="<%= rubygem_version_url(@rubygem, @latest_version.slug) %>" />
+    <link rel="canonical" href="<%= rubygem_version_url(@rubygem, @latest_version.slug, params[:locale] ? { locale: } : {}) %>" />
+    <% I18n.available_locales.each do |_locale| %>
```

Comment:
Should we remove I18n.locale from I18n.available_locales to avoid duplicate links?

Reply:
not per https://developers.google.com/search/docs/specialty/international/localized-versions#all-method-guidelines

> Each language version must list itself as well as all other language versions.